### PR TITLE
KOGITO-3328 Allow using not operator to negate a filter

### PR DESCRIPTION
**Pull request by cristianonicolai**

### CODE REVIEW

**Comment by cristianonicolai** [comment]
@Sara4994 this seems to break the domain explorer, could you please take a look.

**Comment by Sara4994** [comment]
> @Sara4994 this seems to break the domain explorer, could you please take a look.

@cristianonicolai sure

File: persistence-commons/persistence-commons-infinispan/src/main/java/org/kie/kogito/persistence/infinispan/query/InfinispanQuery.java [hunk 1]
```diff
@@ -138,6 +138,8 @@ protected InfinispanQuery(QueryFactory qf, String rootType) {
                     return getRecursiveString(filter, OR);
                 case AND:
                     return getRecursiveString(filter, AND);
+                case NOT:
+                    return format("not %s", filterStringFunction().apply((AttributeFilter<?>) filter.getValue()));
```
Comment (jstastny-cz) [inline persistence-commons/persistence-commons-infinispan/src/main/java/org/kie/kogito/persistence/infinispan/query/InfinispanQuery.java:142]: filter can be even complex expression, right? Do all complex expressions (containing OR, AND) come with surrounding parentheses? Looking at getRecursiveString they probably do, could you confirm?

**Review by jstastny-cz — APPROVED** [review]

File: persistence-commons/persistence-commons-infinispan/src/main/java/org/kie/kogito/persistence/infinispan/query/InfinispanQuery.java (hunk 1 above)
Comment (cristianonicolai) [reply persistence-commons/persistence-commons-infinispan/src/main/java/org/kie/kogito/persistence/infinispan/query/InfinispanQuery.java:142]: @jstastny-cz at this point the support for _not_ operator is for single filter, not complex ones, that need might be something we can revisit in the future if there are use cases for that. For now, I went with a more basic approach that covers the use case needed by the task console.

Comment (pefernan) [reply persistence-commons/persistence-commons-infinispan/src/main/java/org/kie/kogito/persistence/infinispan/query/InfinispanQuery.java:142]: I guess that's fine, if you need grouping conditions inside of a not, you may use an and/or that will add the parenthesis..

**Review by Sara4994 — APPROVED** [review]

**Comment by pefernan** [comment]
@cristianonicolai code looks well but I cannot tested locally!!

**Review by nmirasch — APPROVED** [review]

**Review by pefernan — APPROVED** [review]

**Pull request merged**